gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,593 @@
+      errors.put("message", message);
+      errors.put("reason", reason);
+      args.put("errors", ImmutableList.of(errors));
+      args.put("code", code);
+      args.put("message", message);
+      args.put("status", status);
+      return jsonFactory.toString(ImmutableMap.of("error", args));
+    }
+  }
+
+  private class RequestHandler implements HttpHandler {
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+      // see https://cloud.google.com/resource-manager/reference/rest/
+      String path = exchange.getRequestURI().getPath();
+      String requestMethod = exchange.getRequestMethod();
+      Response response = null;
+      if (requestMethod.equals("POST") && path.startsWith("/" + VERSION + "/projects")) {
+        if (path.contains("undelete")) {
+          try {
+            response = undelete(projectIdFromURI(path));
+          } catch (IOException e) {
+            response = Error.BAD_REQUEST.response(e.getMessage());
+          }
+        } else {
+          String requestBody =
+              decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
+          response = create(jsonFactory.fromString(requestBody, Project.class));
+        }
+      } else if (requestMethod.equals("DELETE")) {
+        response = delete(projectIdFromURI(path));
+      } else if (requestMethod.equals("GET")) {
+        if (path.startsWith("/" + VERSION + "/projects/")) {
+          response = get(projectIdFromURI(path), parseFields(exchange.getRequestURI().getQuery()));
+        } else {
+          response = list(parseListOptions(exchange.getRequestURI().getQuery()));
+        }
+      } else if (requestMethod.equals("PUT")) {
+        String requestBody = decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
+        response = replace(jsonFactory.fromString(requestBody, Project.class));
+      }
+      response = firstNonNull(
+          response, Error.BAD_REQUEST.response("The server could not understand the request."));
+      exchange.getResponseHeaders().set("Content-type", "application/json; charset=UTF-8");
+      exchange.sendResponseHeaders(response.code(), response.body().length());
+      OutputStream outputStream = exchange.getResponseBody();
+      outputStream.write(response.body().getBytes(StandardCharsets.UTF_8));
+      outputStream.close();
+    }
+  }
+
+  private static String decodeContent(Headers headers, InputStream inputStream) throws IOException {
+    List<String> contentEncoding = headers.get("Content-encoding");
+    InputStream input = inputStream;
+    if (contentEncoding != null && contentEncoding.size() > 0
+        && contentEncoding.get(0).contains("gzip")) {
+      input = new GZIPInputStream(inputStream);
+    }
+    return new String(ByteStreams.toByteArray(input), StandardCharsets.UTF_8);
+  }
+
+  private static String projectIdFromURI(String path) throws IOException {
+    String[] pathSplit = path.split("/");
```

Comment:
Consider my suggestion for the context which would hopefully simplify this.

If not, another option would be to have as a static constant BASE_URI with ("/v1beta1/projects/" created using the version constant) and you can use URI [function](http://docs.oracle.com/javase/6/docs/api/java/net/URI.html#relativize(java.net.URI)) to get the extra.